### PR TITLE
Move grenade components to shared

### PR DESCRIPTION
**Pull request by Aexxie**
## About the PR

Moves a ton of grenade components to shared

## Why / Balance

Involved with whitelisted item issues with gridinv

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

FlashComponent, FlashOnTriggerComponent, SmokeOnTriggerComponent, ExplodeOnTriggerComponent, OnUseTimerComponent, and ActiveTimerTriggerComponent all were moved to Shared

:cl: Aexxie
- fix: Flashes, Flashbangs, Teargas, and some other items now can be put on secbelts again.

### CODE REVIEW

File: Content.Client/Explosion/SmokeOnTriggerSystem.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿using Content.Shared.Explosion.EntitySystems;
+
+namespace Content.Client.Explosion;
+
+public sealed class SmokeOnTriggerSystem: SharedSmokeOnTriggerSystem
+{
+	
+}
```
Comment (DrSmugleaf) [inline Content.Client/Explosion/SmokeOnTriggerSystem.cs:8]:
```suggestion
public sealed class SmokeOnTriggerSystem : SharedSmokeOnTriggerSystem
{
}
```

File: Content.Shared/Explosion/Components/ActiveTimerTriggerComponent.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+using Robust.Shared.Audio;
+
+namespace Content.Shared.Explosion.Components;
+
+/// <summary>
+///     Component for tracking active trigger timers. A timers can activated by some other component, e.g. <see cref="OnUseTimerTriggerComponent"/>.
+/// </summary>
+[RegisterComponent]
```
Comment (DrSmugleaf) [inline Content.Shared/Explosion/Components/ActiveTimerTriggerComponent.cs:8]:
```suggestion
using Robust.Shared.Audio;
using Robust.Shared.GameStates;

namespace Content.Shared.Explosion.Components;

/// <summary>
///     Component for tracking active trigger timers. A timers can activated by some other component, e.g. <see cref="OnUseTimerTriggerComponent"/>.
/// </summary>
[RegisterComponent, NetworkedComponent]
```

File: Content.Shared/Explosion/Components/OnUseTimerTriggerComponent.cs [hunk 3]
```diff
@@ -1,57 +1,48 @@
 using Robust.Shared.Audio;
```
Comment (DrSmugleaf) [inline Content.Shared/Explosion/Components/OnUseTimerTriggerComponent.cs:1]:
```suggestion
using Robust.Shared.Audio;
using Robust.Shared.GameStates;
```

File: Content.Shared/Explosion/Components/OnUseTimerTriggerComponent.cs [hunk 4]
```diff
@@ -1,57 +1,48 @@
...
 
-namespace Content.Server.Explosion.Components
+namespace Content.Shared.Explosion.Components
 {
     [RegisterComponent]
```
Comment (DrSmugleaf) [inline Content.Shared/Explosion/Components/OnUseTimerTriggerComponent.cs:5]:
```suggestion
    [RegisterComponent, NetworkedComponent]
```

File: Content.Shared/Explosion/EntitySystems/SharedSmokeOnTriggerSystem.cs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Content.Shared.Explosion.EntitySystems;
+
+public abstract class SharedSmokeOnTriggerSystem: EntitySystem
```
Comment (DrSmugleaf) [inline Content.Shared/Explosion/EntitySystems/SharedSmokeOnTriggerSystem.cs:3]:
```suggestion
public abstract class SharedSmokeOnTriggerSystem : EntitySystem
```

File: Content.Shared/Explosion/EntitySystems/SharedTriggerSystem.cs [hunk 6]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Content.Shared.Explosion.EntitySystems;
+
+public abstract class SharedTriggerSystem: EntitySystem
```
Comment (DrSmugleaf) [inline Content.Shared/Explosion/EntitySystems/SharedTriggerSystem.cs:3]:
```suggestion
public abstract class SharedTriggerSystem : EntitySystem
```

File: Content.Shared/Explosion/Components/OnTrigger/ExplodeOnTriggerComponent.cs [hunk 7]
```diff
@@ -1,4 +1,4 @@
-namespace Content.Server.Explosion.Components;
+namespace Content.Shared.Explosion.Components.OnTrigger;
```
Comment (DrSmugleaf) [inline Content.Shared/Explosion/Components/OnTrigger/ExplodeOnTriggerComponent.cs:1]: This needs to be NetworkedComponent too

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]

**Comment by Aexxie** [comment]
Changed changelog because I somehow forgot about the whole reason I made this for.

File: Content.Shared/Flash/Components/FlashOnTriggerComponent.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Flash.Components
+{
```
Comment (0x6273) [inline Content.Shared/Flash/Components/FlashOnTriggerComponent.cs:4]: Use file-scoped namespace

File: Content.Shared/Explosion/Components/OnUseTimerTriggerComponent.cs [hunk 9]
```diff
@@ -1,57 +1,49 @@
...
-namespace Content.Server.Explosion.Components
+namespace Content.Shared.Explosion.Components
 {
-    [RegisterComponent]
+    [RegisterComponent, NetworkedComponent]
     public sealed partial class OnUseTimerTriggerComponent : Component
     {
-        [DataField("delay")]
-        public float Delay = 1f;
+        [DataField("delay")] public float Delay = 1f;
```
Comment (0x6273) [inline Content.Shared/Explosion/Components/OnUseTimerTriggerComponent.cs:9]: Why change this? In most files it's on separate lines.

**Review by 0x6273 — COMMENTED** [review]
image

Some of the files have wrong line endings. You need to configure your editor/IDE to use the settings in `.editorconfig`, and then resave those files.

File: Content.Shared/Explosion/Components/OnUseTimerTriggerComponent.cs (hunk 9 above)
Comment (Aexxie) [reply Content.Shared/Explosion/Components/OnUseTimerTriggerComponent.cs:9]: > Why change this? In most files it's on separate lines.

Old formatting mistake.

**Pull request merged**